### PR TITLE
Minor changes for property search algo

### PR DESCRIPTION
**Pull request by gayathrir11**
## Summary

#1294

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://user-images.githubusercontent.com/87973126/178913405-e177456d-dcc1-4c85-8605-d7ed95855200.mp4

### CODE REVIEW

**Comment by gayathrir11** [comment]
@akphi please take a look. Addressed things which needs to be fixed so that we can do a release. Will improve the things we discussed in a later commit next week

File: .changeset/yellow-needles-love.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+"@finos/legend-query": patch
+---
```
Comment (MauricioUyaguari) [inline .changeset/yellow-needles-love.md:3]: if we are improving the algo, i think we should include a changeset of what we are doing

File: packages/legend-query/src/stores/QueryBuilderPropertySearchPanelState.ts [hunk 2]
```diff
@@ -297,11 +293,15 @@ export class QueryBuilderPropertySearchPanelState {
 
   fetchMappedPropertyNodes(propName: string): void {
     const propertyName = propName.toLowerCase();
-    this.allMappedPropertyNodes.forEach((node) => {
+    for (let i = 0; i < this.allMappedPropertyNodes.length; i++) {
+      const node = guaranteeNonNullable(this.allMappedPropertyNodes[i]);
       if (node.label.toLowerCase().includes(propertyName)) {
         this.searchedMappedPropertyNodes.push(node);
+        if (this.searchedMappedPropertyNodes.length > 100) {
```
Comment (MauricioUyaguari) [inline packages/legend-query/src/stores/QueryBuilderPropertySearchPanelState.ts:300]: 100 should probably be constant somewhere

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: .changeset/yellow-needles-love.md (hunk 1 above)
Comment (gayathrir11) [reply .changeset/yellow-needles-love.md:3]: Not really improving the algo. That would be a later commit though. May be i should just change the title

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**